### ISSUE TITLE
TV does not work

### ISSUE THREAD

**Issue by Moltresrider** [ACGaming/MrCrayfishFurnitureModRebuilt#4]
**Describe the bug**
The issue persists from the original Mr. Crayfish mod. The TV either displays an untrusted domain error or a not a gif error

**To Reproduce**
If applicable, the steps to reproduce the behavior:
1. Place Modern TV
2. Right click TV
3. Add URL: https://i.imgur.com/GpFlgWu.gif
4. See error: Untrusted Domain

**Screenshots**
![2024-08-10_08 38 28](https://github.com/user-attachments/assets/2b2cd7ce-0386-42e1-abf7-906108f946c9)

![2024-08-10_08 52 52](https://github.com/user-attachments/assets/c8287340-3458-4ec6-9f2e-11b69b9b2510)

Please complete the following information:
- Minecraft Version: 1.12.2
- Mod Version: 6.5.0

**Comment by xJon** [comment]
Maybe relevant to [this](https://github.com/Kanzaji/XU2-Patcher/blob/master/src/xu2/patched/patches/1.12/com/rwtema/extrautils2/textures/ImgurTextureHolder.java.patch)?

**Comment by ACGaming** [comment]
Yeah, it's possibly related to an invalid/outdated user agent string.

**Comment by Ketei** [comment]
Had a similar issue. Images and GIFs were failing to load from specific HTTPS sites (e.g., Wikipedia) but loading correctly from others. Tried building with newer user-agent and even custom ones to no avail.

For context, I'm using the official Minecraft Launcher, which bundles its own JRE per version including a set of trusted root certificates. So the bundled version of the JRE for the 1.12.2 is going to be older than the one bundled for 1.21.5.

After some testing I found out that the reason I wasn't able to get some images was because my launcher's bundled java certification entries were old/expired (and some probably missing) resulting in the error `sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target` which makes sense since the 1.12.2 version is old by now.

tl;dr
To fix it I made a copy of the certifications (cacerts file) of the latest version of minecraft (1.21.5 as of writing this) and using java arguments on the launcher to point the 1.12.2 installation to use the newer cacerts file.

**Comment by xJon** [comment]
@Ketei I think that LoliASM has a certificate fix for old Java? https://github.com/LoliKingdom/LoliASM/commit/8f76f3016b1945e6dbd5747e0c74bf01652139e5

**Comment by Ketei** [comment]
Thank you for your reply @xJon. I was testing with only this mod active and I never tested alongside with LoliASM, but I can confirm that the mod does indeed fix the certificate issues. So it's probably easier to just install LoliASM/CensoredASM than using new certificates.

The more options the better.

**Comment by meh8-ui** [comment]
> [@Ketei](https://github.com/Ketei) I think that LoliASM has a certificate fix for old Java? [LoliKingdom/LoliASM@8f76f30](https://github.com/LoliKingdom/LoliASM/commit/8f76f3016b1945e6dbd5747e0c74bf01652139e5)

that mod keeps crashing unfortunately. Basically minecraft won't even open with that mod :( i can't use it and gifs from tenor just don't work on tv in game =(
